File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/crypto/impl/NoOpCryptoService.java
package at.tugraz.tc.cyfile.crypto.impl;

import java.security.InvalidKeyException;

import at.tugraz.tc.cyfile.crypto.CryptoService;

/**
 * Implementation of {@link CryptoService}
 * <p>
 * that does nothing
 * Created by cobri on 3/21/2018.
 */

public class NoOpCryptoService implements CryptoService {
    @Override
    public byte[] encrypt(byte[] data) {
        return data;
    }

    @Override
    public String encrypt(String data) { return data; }

    @Override
    public byte[] decrypt(byte[] cipherData) {
        return cipherData;
    }

    @Override
    public String decrypt(String cipherData) {
        return cipherData;
    }

}

File: CyFile-Android/app/src/main/kotlin/at/tugraz/tc/cyfile/crypto/impl/NoOpCryptoService.kt
package at.tugraz.tc.cyfile.crypto.impl

import at.tugraz.tc.cyfile.crypto.CryptoService

/**
 * Implementation of [CryptoService]
 *
 *
 * that does nothing
 * Created by cobri on 3/21/2018.
 */

class NoOpCryptoService : CryptoService {
    override fun encrypt(data: ByteArray): ByteArray {
        return data
    }

    override fun encrypt(data: String): String {
        return data
    }

    override fun decrypt(cipherData: ByteArray): ByteArray {
        return cipherData
    }

    override fun decrypt(cipherData: String): String {
        return cipherData
    }

}
